tests/system/src/test/java/test/javafx/stage/SystemMenuBarTest.java

```diff
@@ -108,4 +113,61 @@ public MenuBar createFailingMenuBar() {
 
         return menuBar;
     }
+
+    @Test
+    public void testMemoryLeak() throws InterruptedException {
+        Util.runAndWait(() -> {
+            Thread.currentThread().setUncaughtExceptionHandler((t,e) -> {
+                e.printStackTrace();
+                failed.set(true);
```

Comment:
To avoid the test hanging indefinitely, I recommend adding the following after setting the `failed` flag:

```
    memoryLatch.countDown();
```

Reply:
good idea.